### ISSUE TITLE
Better name for deferred

### ISSUE THREAD

**Issue by johanandren** [akka/akka#24548]
`Behaviors.deffered` though perhaps correct is somewhat hard to find/understand, (discussion on gitter: https://gitter.im/akka/dev?at=5a74c5bb7dcd63481f0e04a4 ), we should rename it t something that more clearly express what it does, current best bet is `onStart`

**Comment by chbatey** [comment]
How about just immutable and it is an overload? It produces an immutable but gives you a ctx to do stuff before starting and doesn't tie us down to how lazy it is / will ever be.

**Comment by patriknw** [comment]
but it can and should be able to return any behavior, and I bet that types go bad with an overload

**Comment by patriknw** [comment]
in similar thinking I was suggesting `create`, but I guess that was not popular
